From b75ceb520acb46839f984b1ca5f9579d8914cef0 Mon Sep 17 00:00:00 2001
From: Daniel Charles <daniel.charles@intel.com>
Date: Wed, 23 Oct 2013 14:34:52 +0200
Subject: Change max width and height supported by H.263 decoder.

Currently this is set to CIF PAL resolution (352x288) which
won't calculate the actual output buffer size for higher resolutions

The h.263 decoder needs the max width and height supported to be able
to decode

H.263 decoders supports: QCIF, CIF, D1 and 16CIF; all PAL.

Change-Id: I38664f1fcfbda2b73f3fdc5aca719cbaee803f0e
Signed-off-by: Daniel Charles <daniel.charles@intel.com>
---
 .../omx/SoftVideoDecoderOMXComponent.cpp           |    7 +++++++
 1 files changed, 7 insertions(+), 0 deletions(-)

diff --git a/media/libstagefright/omx/SoftVideoDecoderOMXComponent.cpp b/media/libstagefright/omx/SoftVideoDecoderOMXComponent.cpp
index 08a3d42..00a04da 100644
--- a/media/libstagefright/omx/SoftVideoDecoderOMXComponent.cpp
+++ b/media/libstagefright/omx/SoftVideoDecoderOMXComponent.cpp
@@ -114,6 +114,13 @@ void SoftVideoDecoderOMXComponent::initPorts(
 
     addPort(def);
 
+    if (!strcasecmp("video/3gpp", const_cast<char *>(mimeType))) {
+        // H.263 requires the max size supported by the decoder
+        // which is 16CIF (1408x1152)
+        mWidth=1408;
+        mHeight=1152;
+    }
+
     updatePortDefinitions();
 }
 
-- 
1.7.4.1

